default/python/AI/fleet_orders.py

```diff
@@ -527,11 +556,16 @@ def is_valid(self):
 
     def issue_order(self):
         if not super(OrderRepair, self).issue_order():
-            return
+            return False
         fleet_id = self.fleet.id
         system_id = self.target.get_system().id
-        fleet = self.fleet.get_object()
-        if system_id not in [fleet.systemID, fleet.nextSystemID]:
+        fleet = self.fleet.get_object()  # type: fo.fleet
+        if system_id == fleet.systemID:
+            if foAI.foAIstate.get_fleet_role(fleet_id) == MissionType.EXPLORATION:
+                if system_id in foAI.foAIstate.needsEmergencyExploration:
+                    foAI.foAIstate.needsEmergencyExploration.remove(system_id)
+        elif system_id != fleet.nextSystemID:
+            self.executed = False
```

Comment:
This assignment is what made me confused about the behavior of the function. It suggests that the order is only executed after moving to the target system. However, it will always be overwritten at the end of function.

Can we remove this or is this necessary for logging purposes (please add a short comment in that case)?

Reply:
> Can we remove this

Done.  I overlooked cleaning that up when I added the superseding health check.